app/main.py

```diff
@@ -1,25 +1,67 @@
-from abc import ABC
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.private_name = "_" + name
+
+    def __get__(self, obj: object, objtype: type = None) -> int:
+        return getattr(obj, self.private_name)
+
+    def __set__(self, obj: object, value: int) -> None:
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError(f"{value} is out of range "
+                             f"({self.min_amount}, {self.max_amount})")
+        setattr(obj, self.private_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self,
+                 name: str,
+                 age: int,
+                 weight: int,
+                 height: int
+                 ) -> None:
+        self.name = name
+        self.weight = weight
+        self.height = height
+        self.age = age
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    @staticmethod
+    @abstractmethod
+    def validate(visitor: Visitor) -> bool:
+        pass
```

Comment:
There is no need for `valid` methods in all classes, you should define `age`, `weight` and `height` as class attributes and assign instances of IntegerRange class with limitations to each range. It will allow to utilize the `__set__` method later to trigger an error for further processing.